### PR TITLE
Update 'devbuild.sh', module files, and directory names

### PR DESCRIPTION
**Pull request by chan-hoo**
## DESCRIPTION OF CHANGES: 
- Updates `devbuild.sh` and adds `devclean.sh`
- Changes the directory names: bin -> exec, src -> sorc
- Moves the module files to `modulefiles` and updates modules with the latest ones.
- Updates the hash of UFS_UTILS.

## TESTS CONDUCTED: 
- Machines/platforms: wcoss2 (cactus), hera, orion

## ISSUE: 
- Partially fixes the issue(s) mentioned in #3

### CODE REVIEW

**Comment by chan-hoo** [comment]
@MatthewPyle-NOAA, could you give the code owners (listed in `.github/CODEOWNERS`) the write access to this repository? Then, they will be set as the default PR reviewers when a new PR is submitted. I can see the error message `This CODEOWNERS file contains errors`. This is because they don't have the write access. If you think the file (.github/CONDEOWNERS) needs to be modified, please let me know.

File: README.md
Comment (chan-hoo) [inline README.md:23]: As can be seen in this new README file, this PR makes the build steps work up to Step 3 on `wcoss2`, `hera` and `orion`.

Comment (ShunLiu-NOAA) [reply README.md:23]: @chan-hoo Thank you so much for working on this so quickly. Since dev-sci will be used by both GSL and EMC, I add @hu5970 as a reviewer so that GSL can get an understanding of how the modification will affect the future update of RRFS-B on JET.

**Comment by ShunLiu-NOAA** [comment]
@hu5970 Please review the modifications and let's consider the new development in dev-sci.

File: README.md
Comment (chan-hoo) [reply README.md:23]: Thanks. As described above, the rrfs-workflow does not work yet. This PR only makes the 'build' step work well. I've tested it on wcoss2, hera, and orion. I don't have access to Jet. Are there any other machines to be tested?

Comment (BenjaminBlake-NOAA) [reply README.md:23]: @chan-hoo  I think the RRFS workflow for operations will only need to be supported on WCOSS2, Hera, Jet, and Orion.  So at some point we will want to remove the logic for the other non-supported platforms.

**Comment by MatthewPyle-NOAA** [comment]
> @MatthewPyle-NOAA, could you give the code owners (listed in `.github/CODEOWNERS`) the write access to this repository? Then, they will be set as the default PR reviewers when a new PR is submitted. I can see the error message `This CODEOWNERS file contains errors`. This is because they don't have the write access. If you think the file (.github/CONDEOWNERS) needs to be modified, please let me know.

@chan-hoo I'll try to get this sorted out

**Comment by BenjaminBlake-NOAA** [comment]
@chan-hoo  I am testing the build of rrfs-workflow on WCOSS2 and Hera now.  I have access to Jet (although I've never done anything on that machine) and will also try building it there.

**Comment by BenjaminBlake-NOAA** [comment]
Does anyone know what directory I can use on Jet to test building the workflow?  I'm not familiar with the directory structure on Jet, and my home directory won't work because the disk quota is too small.

File: .github/CODEOWNERS
Comment (chan-hoo) [inline .github/CODEOWNERS:6]: @MatthewPyle-NOAA, I've updated the codeowners, but it still says "unknown owner".

Comment (MatthewPyle-NOAA) [reply .github/CODEOWNERS:6]: Yeah, I saw that.  Is there a need for the codeowners file since we have github level control of the team?  Can you see this list of developers (which should be able to issue and interact with PRs and other normal non-admin github activity) [https://github.com/orgs/NOAA-EMC/teams/rrfs-devs](https://github.com/orgs/NOAA-EMC/teams/rrfs-devs)

File: ush/bash_utils/define_macos_utilities.sh
Comment (MatthewPyle-NOAA) [inline ush/bash_utils/define_macos_utilities.sh:1]: Is this macos utility defining file needed with the machines we're supporting with the RRFS workflow?

File: ush/bash_utils/eval_METplus_timestr_tmpl.sh
Comment (MatthewPyle-NOAA) [inline ush/bash_utils/eval_METplus_timestr_tmpl.sh:1]: Another item that we'll want to remove in the long run (not running metplus within the RRFS workflow).

File: ush/constants.sh
Comment (MatthewPyle-NOAA) [inline ush/constants.sh:1]: Where does this "radius_Earth" get used?  Just wondering if it is important, and if it is consistent with radius values used in component codes like the model and post

File: ush/setup.sh
Comment (MatthewPyle-NOAA) [inline ush/setup.sh:1]: I'm assuming the permission change (644 --> 755) was intentional?

**Review by MatthewPyle-NOAA — COMMENTED** [review]
@chan-hoo Changes look good to me.  I left a few comments, mostly about stuff that I think needs to be removed (JEDI, metplus, etc.), but maybe not critical to fix at this stage.

File: jobs/JREGIONAL_RUN_PYTHON_GRAPHICS
Comment (chan-hoo) [reply jobs/JREGIONAL_RUN_PYTHON_GRAPHICS:1]: Got it.

File: ush/bash_utils/define_macos_utilities.sh
Comment (chan-hoo) [reply ush/bash_utils/define_macos_utilities.sh:1]: I agree. I'll remove this file when Dogwood is back to us.

File: ush/bash_utils/eval_METplus_timestr_tmpl.sh
Comment (chan-hoo) [reply ush/bash_utils/eval_METplus_timestr_tmpl.sh:1]: Will remove this file too when Dogwood is available.

File: ush/constants.sh
Comment (chan-hoo) [reply ush/constants.sh:1]: As far as I know, this is used by UFS_UTILS (maybe ESG grid). In this PR, the hash of the UFS_UTILS is updated with the latest one. So I think this value should be updated too (this value is what the UFS SRW App uses).

File: ush/setup.sh
Comment (chan-hoo) [reply ush/setup.sh:1]: I changed this for testing. I am not sure whether or not this is necessary. Do you think 644 is better?

Comment (MatthewPyle-NOAA) [reply ush/setup.sh:1]: Makes sense to me for it being an executable - no need to change it.

File: ush/constants.sh
Comment (MatthewPyle-NOAA) [reply ush/constants.sh:1]: I spot checked the UPP and regional_esg_grid codes, and saw the same 6371.2 km value, so I'll close this comment.

File: ush/bash_utils/define_macos_utilities.sh
Comment (chan-hoo) [reply ush/bash_utils/define_macos_utilities.sh:1]: Removed.

File: ush/bash_utils/eval_METplus_timestr_tmpl.sh
Comment (chan-hoo) [reply ush/bash_utils/eval_METplus_timestr_tmpl.sh:1]: Removed.

**Review by MatthewPyle-NOAA — DISMISSED** [review]
No objections from me from looking at changes...but hopefully Ben Blake can weigh in after more testing.

**Comment by BenjaminBlake-NOAA** [comment]
@MatthewPyle-NOAA @chan-hoo  I can definitely look over all these changes more thoroughly tomorrow morning.  The build process worked well for me on WCOSS2 and Hera.  Once I get access to the wrfruc project on Jet I will test building the code there and submit my final approval.  Hopefully that won't take more than a day or two!

**Comment by chan-hoo** [comment]
Sorry. Something wrong. I'll fix the dismiss and label issue.

**Comment by chan-hoo** [comment]
@MatthewPyle-NOAA, I don't know what happened here. I didn't dismiss your approval. Can you fix this?

**Review by MatthewPyle-NOAA — APPROVED** [review]
Giving conditional approval, pending Ben Blake's testing.

**Comment by chan-hoo** [comment]
> Giving conditional approval, pending Ben Blake's testing.

Thank you!

File: .github/CODEOWNERS
Comment (BenjaminBlake-NOAA) [inline .github/CODEOWNERS:6]: I know some work was done to update the CODEOWNERS file, but GitHub is still saying the file contains errors.  Should we revert back to the previous version which listed our individual usernames?

File: devclean.sh
Comment (BenjaminBlake-NOAA) [inline devclean.sh:7]: I would suggest replacing UFS-SRW application with RRFS here.

File: etc/lmod-setup.csh
Comment (BenjaminBlake-NOAA) [inline etc/lmod-setup.csh:9]: I would remove the reference to Cheyenne.

File: etc/lmod-setup.sh
Comment (BenjaminBlake-NOAA) [inline etc/lmod-setup.sh:10]: I would also remove this reference to Cheyenne.

File: ush/load_modules_wflow.sh
Comment (BenjaminBlake-NOAA) [inline ush/load_modules_wflow.sh:19]: I would remove Cheyenne as an option here as well.

**Review by BenjaminBlake-NOAA — COMMENTED** [review]
@chan-hoo I've finished reviewing all of the changes, they look great!  I have a couple of general comments which I will leave here, and I added a few other suggestions to specific files.  I am not going to formally submit my approval just yet but I will do that once I get access to Jet and can build the workflow there.

- I agree with Matt about removing JEDIENVAR_IODA and RUN_PYTHON_GRAPHICS, but that can be done later.
- Once we start running/testing the workflow and jobs, we should change python_srw to python_rrfs and SRW_ENV to RRFS_ENV in modulefiles/tasks.  But I think that change is outside the scope of this PR.

**Review by BenjaminBlake-NOAA — APPROVED** [review]

**Comment by MatthewPyle-NOAA** [comment]
Are you ready for this to be merged, @chan-hoo ?

**Comment by chan-hoo** [comment]
@MatthewPyle-NOAA, whenever I submit a new commit, the existing reviewer's approvals are dismissed. I think we need to change some settings.

File: .github/CODEOWNERS
Comment (chan-hoo) [reply .github/CODEOWNERS]: Done

File: devclean.sh
Comment (chan-hoo) [reply devclean.sh]: Done

File: etc/lmod-setup.csh
Comment (chan-hoo) [reply etc/lmod-setup.csh]: Done

File: etc/lmod-setup.sh
Comment (chan-hoo) [reply etc/lmod-setup.sh]: Done

**Comment by chan-hoo** [comment]
I found out your approval was only dismissed by a new commit. I think the code owner' (new) review is required for any new commits.

**Comment by MatthewPyle-NOAA** [comment]
@chan-hoo I found this [issue](https://github.com/isaacs/github/issues/1157), but don't see the option described to stop the dismissal of reviews.  I'll reapprove.

**Review by MatthewPyle-NOAA — APPROVED** [review]
Giving another approval...

**Comment by chan-hoo** [comment]
Thanks, I think it is ready to be merged now.

**Pull request merged**
